playbooks/user-password.yml

```diff
@@ -0,0 +1,33 @@
+---
+# Generate password for user
+# You can also generate a password with `mkpasswd -m sha-512`
+- hosts: all
+  become: true
+  gather_facts: false
+  vars:
+    pwgen: "{{ lookup('password', '/dev/null length=8 chars=ascii_letters') }}"
+  tasks:
+    - set_fact:
+        passwd: "{{ pwgen }}"
```

Comment:
Maybe I don't understand the value of facts yet, but why not simply merge all this into the users section below?
```
users:
  password: "lookup('password', '/dev/null length=8 chars=ascii_letters') | password_hash('sha512', 'salty') }}"
```
Also, I think you can get some free extra security by using e.g. the hostname as the salt instead of something static.

Reply:
I can't print the password that way. Really we should be using the vault